docs/realtime.md

```diff
@@ -5,16 +5,26 @@ title: Realtime Responsiveness Evaluation
+
+An evaluation of the pipeline latency with a single `InferenceCore` thread (see [below](#frame-rate-improvement-by-using-multi-threading)) revealed a ~180ms latency when running on a laptop and ~890ms latency on the Raspberry Pi 4B. These figures match the requirements we have set out for the system. They also indicate that users who have more stringent performance requirements should use PosturePerfection on laptop or desktop machines as opposed to low-power single-board computers like the Raspberry Pi.
 
 ## Frame Rate Improvement by using Multi-threading
 
-During initial development, the pipeline did not make use of multi-threading and therefore resulted in a very slow frame rate and high latency in processing a video stream. Therefore a major change was implemented to exploit multi-threading. As mentioned above, the post processing stage must be performed on frames in the correct order due to the use of `IIR` filtering. However, the pose estimation stage can be parallelised to significantly improve performance. Instead of waiting for the stage to process a frame before another can be processed, another waiting thread can process the given frame. Below is an evaluation of how a higher number of threads for this middle stage results in a higher frame rate being able to be used:
+Operating all stages of the pipeline will yields poor performance as each stage is forced to wait until processing from the previous stage is completed. An inspection of execution times for the individual pipeline stages reveals the major bottleneck of the system to be the `InferenceCore`, which takes hundreds of milliseconds to execute for a single frame. It makes sense, therefore, to split this stage into a separate thread to allow other stages to run concurrently. In our pipeline we have opted for this approach, with the three stages (frame-capture, inference, interpretation of results) running in separate threads. We have an added requirement for the system that frames must be passed from one stage to the next in order. This has two reasons: 1. the output should be in order, 2. internal processing (including the IIR filter) requires frames in-order fro outcomes to be sensible.
```

Comment:
"requires frames in-order ~fro~ outcomes" -> "requires frames in-order for outcomes"